### PR TITLE
Add ARM support to the Azure pack

### PR DESCRIPTION
**Pull request by tonybaloney**
I told someone that StackStorm's Azure pack supports Azure Resource Manager (ARM), turns out it doesn't. Adding support so I don't look silly

### CODE REVIEW

File: packs/azure/actions/lib/base.py [hunk 1]
```diff
@@ -40,3 +41,16 @@ def _get_driver(self, name, access_key):
         cls = get_storage_driver(StorageProvider.AZURE_BLOBS)
         driver = cls(key=name, secret=access_key)
         return driver
+
+
+class AzureBaseResourceManagerAction(Action):
+    def __init__(self, config):
+        super(AzureBaseStorageAction, self).__init__(config=config)
+
+        from msrestazure.azure_active_directory import ServicePrincipalCredentials
```
Comment (manasdk) [inline packs/azure/actions/lib/base.py]: oh no!

Comment (tonybaloney) [inline packs/azure/actions/lib/base.py]: fixed.

File: packs/azure/requirements.txt [hunk 2]
```diff
@@ -1 +1,2 @@
+azure-mgmt==0.30.0a1
```
Comment (manasdk) [inline packs/azure/requirements.txt]: Hmm, an alpha release. I see everything they have is an alpha on pypi

File: packs/azure/config.yaml [hunk 3]
```diff
@@ -5,3 +5,7 @@ compute:
 storage:
   name:
   access_key:
+resource_manager:
+  client_id: 'ABCDEFGH-1234-ABCD-1234-ABCDEFGHIJKL'
```
Comment (manasdk) [inline packs/azure/config.yaml]: To be consistent can you blank these out?

**Comment by manasdk** [comment]
+1

Will merge after travis completes.

**Comment by tonybaloney** [comment]
Green lights! All done @manasdk

**Comment by manasdk** [comment]
Great!

Thanks for the contribution :)

**Pull request merged**